src/main/scala/org/apache/spark/sql/execution/datasources/greenplum/GreenplumUtils.scala

```diff
@@ -97,9 +103,28 @@ object GreenplumUtils extends Logging {
       }
     }
 
-    df.foreachPartition { rows =>
+    var finalDf = df
+    if (isAppend) {
+      // TODO: For the case, which should append data to table, is there more efficient method?
+      finalDf = df.coalesce(1)
+    }
+
+    val partNum = finalDf.rdd.getNumPartitions
+    val accumulator = finalDf.sparkSession.sparkContext.longAccumulator("copySuccess")
+    val randomString = UUID.randomUUID().toString.flatMap {
```

Comment:
Why should filter “-” character？
And why not just use filter or filterNot func？

Reply:
For that the '-' is  an invalid symbol for table name.